### ISSUE TITLE
Auto-loaded external audio tracks should not take precedence

### ISSUE THREAD

**Issue by XanClic** [mpv-player/mpv#3777]
mpv 0.21.0 / Arch Linux

### Actual behavior

```--audio-file-auto=exact``` is the default and automatically loaded files behave like files specified explicitly with ```--audio-file```, i.e. their contents take precedence over the audio tracks in the video file. Therefore, if the user does not know about ```--audio-file-auto```, it is possible for an unrelated audio file to unexpectedly change the audio of some video file in the same directory.

In my case, for instance, I had the two files “out.mkv” and “out.flac”, both of which were just temporary test outputs of some ffmpeg commands.

### Expected behavior

Audio tracks loaded automatically from external files should not take precedence over tracks from the video file itself. Otherwise, mpv is basically requiring users to know about the existence of ```--audio-file-auto``` and that it is turned on by default.

Cheers,
Max

**Comment by Hrxn** [comment]
It's not a bug, it's a feature! ™®©

**Comment by wiiaboo** [comment]
MPC-HC also does this. You shouldn't use the same name for both if they're unrelated.

**Comment by XanClic** [comment]
Hrxn, and it's a good feature. But in my opinion, it's not a good default.

wiiaboo, there were not completely unrelated. out.flac was indeed to be used as the audio track of out.mkv, but with an offset set through ffmpeg. This is why it took me some time to figure out mpv as the “culprit”, because while the correct sound was playing, it had the wrong offset and I thought my ffmpeg invocation was somehow wrong.

It's easy to say “You shouldn't use the same name” and also “Well, then just put ```audio-file-auto=no``` into your config file” (which I have now), but both don't make any sense if you don't know about this behavior. And this is my point: To me, the feature seems very useful, but it's weird to be the default. When someone wants external audio tracks to load automatically, they are more likely to know or expect such an option to exist and can thus enable it, in contrast to someone who doesn't want this.

Max

**Comment by lachs0r** [comment]
I’ve disabled this option too because I ran into the exact same situation with exactly the same file names as OP :)
I think autoloaded external files should *never* take precedence over existing tracks of the media file that is being played. This goes for subtitles as well. It’s just not the behavior I would expect, especially with audio.

**Comment by wiiaboo** [comment]
I expect external subs to take precedence if fuzziness is enabled and it's the same language. I've never used external audio tracks, so I don't expect anything.

**Comment by lachs0r** [comment]
Sigh. Perhaps our option list needs to be a bit longer still. `--[sub|audio-file]-auto-select=[always|same-language|never]`? How to detect the language of external files? They’re almost never tagged. Please don’t suggest looking at the file name.

**Comment by wiiaboo** [comment]
`--sub-auto=fuzzy` uses the languages codes in `--slang`/`--alang`, iirc. So "movie.eng.srt" or "movie.en.srt" is considered English and would trump an internal track tagged as English language.

**Comment by GeneralFailer** [comment]
I've already tried to report this a while ago in #3062.
I've just tried the latest MPC-HC version and it didn't play the external track by default.

**Comment by XanClic** [comment]
GeneralFailer, right, but although I've omitted that in the title (my bad), my main concern is not that external tracks to get priority at all, but that they are prioritized **by default**.
I don't mind ```audio-file-auto=exact``` to prioritize external tracks too much, although I think it would be better not to. What I do mind, however, is that ```audio-file-auto=exact``` is the default, **and** that it prioritizes external tracks.
I think it's completely justified to have an option to auto-load external tracks and let them have precedence, but I don't think that should be the default. If it is too difficult to implement an option to auto-load external tracks without them having precedence, in my very personal opinion ```audio-file-auto=no``` should be the default.

My reasoning is the following: When choosing the default value for an option, one has to ask why the user would not specify that option, and I can imagine two cases:

1. The user is happy with the default.
2. The user does not know about the option.

For (1), the default should be what makes most users happy. For (2), the default should result in the behavior expected by people who don't know that the option exists at all. It is important to note that both are not necessarily the same: Regarding ```--audio-file-auto```, for (1) the default might be ```exact``` and for (2) I think it would be ```no``` (if you don't know about auto-loading of external tracks, you won't expect it to happen).

While it's generally a matter of personal taste which of the cases should get more weight, I very much think that in the case of mpv, (2) should take precedence over (1). This is because you can always override mpv's built-in defaults by putting options into your own config file. Thus, once you learn that ```--audio-file-auto``` exists, you can always just write ```audio-file-auto=exact``` into your config file and be done with it. I think that is much less painful than users having to figure out that the option exists because mpv did something they did not expect.

tl;dr: I'd be happy if ```---audio-file-auto=no``` became the default again, like it was before cfd3d5b52003a23376d02fd6cc581f46289badd2.

Max

**Comment by GeneralFailer** [comment]
@XanClic, your solution wouldn't suit me either, because autoloading is very useful to me even though autoplaying is a serious inconvenience. Setting a new autoload-only option to be default would be a great compromise for both sides IMO,

**Comment by mc4man** [comment]
More users are taking advantage of the current default than those who happen to have a same name external for whatever odd reason. Leave it be & if affected change default  locally.

**Comment by wm4** [comment]
Auto-loading and selecting audio tracks by default was a feature request. But I can revert that if that is more popular.

**Comment by wiiaboo** [comment]
I'm ok for no auto-selection but still auto-loading for audio tracks. Against for sub tracks, for both selection and loading.

**Comment by gitterliu** [comment]
@wm4 Maybe add options like `--autoselect-external-sub` and `--autoselect-external-audio` in the feature? Also there's a bug of `audio-reload` command reloading external audio file with multiple tracks, it should remove all tracks instead of only selected track.

**Comment by doanngochieuo** [comment]
it's kinda funny sometime you back and forth with same issue when u use mpv, it's not that "hard" with core devs when they know the code but it's funny with the idea of force `default` prefer external sub/audio over internal ones.

this makes option like `slang/alang` does not fully functioning.

**Comment by doanngochieuo** [comment]
And then i found out `scripts/*.lua` will solve this issue, write ur lua file to load external stuffs and don't select them automatically, problem solved.

**Comment by doanngochieuo** [comment]
hint how to solve this from user perspective:
1. loop through tracks
2. set sid/aid to 'auto' to prefer embedded track
3. check preferred track list and break when match
4. manage ur own track file to check if the file is loaded before to ignore logic from 1. which mpv will use default track info in watch_later
